src/sentinel.c

```diff
@@ -4061,49 +4061,20 @@ NULL
 
 /* SENTINEL INFO [section] */
 void sentinelInfoCommand(client *c) {
-    
-    char defSections[11][15] = {"server", "clients", "memory", "persistence", "stats", "replication", "cpu", "modules", "errorstats", "cluster", "keyspace"};
-    dict * sections_dict = dictCreate(&setDictType); /* Set to add the subsections to print*/
-
     if (c->argc > 2) {
         addReplyErrorObject(c,shared.syntaxerr);
         return;
     }
 
-    int defsections = 0, allsections = 0, everything = 0;
-    char *section = c->argc == 2 ? c->argv[1]->ptr : NULL;
-    if (section) {
-        allsections = !strcasecmp(section,"all");
-        if (!strcasecmp(section,"default")){
-            for (int i = 0; i < 11; i++){
-                dictAdd(sections_dict, sdsnew(defSections[i]), NULL);
-            }
-            defsections = 1;
-        }
-        everything = !strcasecmp(section,"everything");
-    } else {
-        for (int i = 0; i < 11; i++) {
-                dictAdd(sections_dict, sdsnew(defSections[i]), NULL);
-        }
-        defsections = 1;
-    }
-
-    int sections = 0;
     sds info = sdsempty();
-
-    dictAdd(sections_dict, sdsnew("server"), NULL);
-    dictAdd(sections_dict, sdsnew("clients"), NULL);
-    dictAdd(sections_dict, sdsnew("cpu"), NULL);
-    dictAdd(sections_dict, sdsnew("stats"), NULL);
-
-    info = genRedisInfoString(sections_dict, allsections, everything);
-
-    if (defsections || allsections || !strcasecmp(section,"sentinel")) {
+    info = genRedisInfoString(c, "sentinel");
+    if (c->argc == 1 || !strcasecmp(c->argv[1]->ptr,"all") || !strcasecmp(c->argv[1]->ptr,"default") || !strcasecmp(c->argv[1]->ptr,"sentinel")) {
```

Comment:
this solution won't take "sentinel" in any other argv position (or detect that "all" / "default" was used).
once again, why not take my advise of having the function that populates a dict and returns too booleans, and then this method will trim all elements from that dict except for the few it allows??